### PR TITLE
SyMS Service API Swagger Client

### LINKED ISSUE #18744: [Synapse - SyMS] REST API Review
Service name: **SyMS**
Key contact for review: **piyushdubey (piyushdubey@microsoft.com)**
Whether this is a new or existing API: **New**
Whether or not your service is already been deployed: **Already deployed**
A bried description of the material to be reviewed/changed: **Pull Request - [118742](https://github.com/Azure/azure-rest-api-specs/pull/18742)**

### PR DESCRIPTION
**Pull request by piyushdubey**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [x] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by weidongxu-microsoft** [comment]
@piyushdubey See https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/271/Azure-REST-API-Stewardship

**Comment by piyushdubey** [comment]
@weidongxu-microsoft I don't have the permission to add tags to the issue.
Here's the Issue - https://github.com/Azure/azure-rest-api-specs/issues/18744
I cannot link PR to the issue either. Appreciate your help.

**Review by weidongxu-microsoft — COMMENTED** [review]
| Message | Location |
|---|---|
| Operation might be pageable. Consider adding the x-ms-pageable extension. | [symsAPI.json#L28-L68](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L28-L68) |
| Error response should contain a x-ms-error-code header. | [symsAPI.json#L64-L68](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L64-L68) |
| A 202 response should include an Operation-Location response header. | [symsAPI.json#L111-L113](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L111-L113) |
| Error response should contain a x-ms-error-code header. | [symsAPI.json#L114-L118](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L114-L118) |
| Operation might be pageable. Consider adding the x-ms-pageable extension. | [symsAPI.json#L125-L171](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L125-L171) |
| Error response should contain a x-ms-error-code header. | [symsAPI.json#L167-L171](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L167-L171) |
| Operation might be pageable. Consider adding the x-ms-pageable extension. | [symsAPI.json#L177-L250](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L177-L250) |
| Error response should contain a x-ms-error-code header. | [symsAPI.json#L246-L250](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L246-L250) |
| Operation might be pageable. Consider adding the x-ms-pageable extension. | [symsAPI.json#L256-L336](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L256-L336) |
| Error response should contain a x-ms-error-code header. | [symsAPI.json#L332-L336](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L332-L336) |
| Operation might be pageable. Consider adding the x-ms-pageable extension. | [symsAPI.json#L342-L402](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L342-L402) |
| Error response should contain a x-ms-error-code header. | [symsAPI.json#L398-L402](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L398-L402) |
| Operation might be pageable. Consider adding the x-ms-pageable extension. | [symsAPI.json#L408-L468](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L408-L468) |
| Error response should contain a x-ms-error-code header. | [symsAPI.json#L464-L468](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L464-L468) |
| Operation might be pageable. Consider adding the x-ms-pageable extension. | [symsAPI.json#L474-L541](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L474-L541) |
| Error response should contain a x-ms-error-code header. | [symsAPI.json#L537-L541](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L537-L541) |
| Operation might be pageable. Consider adding the x-ms-pageable extension. | [symsAPI.json#L547-L614](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L547-L614) |
| Error response should contain a x-ms-error-code header. | [symsAPI.json#L610-L614](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L610-L614) |
| Error response should contain a x-ms-error-code header. | [symsAPI.json#L654-L658](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L654-L658) |
| OperationId for put method should contain "Create" or "Update" | [symsAPI.json#L664](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L664) |
| Error response should contain a x-ms-error-code header. | [symsAPI.json#L705-L709](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L705-L709) |
| A delete operation should have a `204` response. | [symsAPI.json#L740-L748](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L740-L748) |
| All success responses except 202 & 204 should define a response body. | [symsAPI.json#L741-L743](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L741-L743) |
| Error response should contain a x-ms-error-code header. | [symsAPI.json#L744-L748](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L744-L748) |
| Error response should contain a x-ms-error-code header. | [symsAPI.json#L815-L819](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L815-L819) |
| OperationId for put method should contain "Create" or "Update" | [symsAPI.json#L825](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L825) |
| Error response should contain a x-ms-error-code header. | [symsAPI.json#L908-L912](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L908-L912) |
| A delete operation should have a `204` response. | [symsAPI.json#L970-L978](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L970-L978) |
| All success responses except 202 & 204 should define a response body. | [symsAPI.json#L971-L973](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L971-L973) |
| Error response should contain a x-ms-error-code header. | [symsAPI.json#L974-L978](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L974-L978) |
| Error response should contain a x-ms-error-code header. | [symsAPI.json#L1052-L1056](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L1052-L1056) |
| OperationId for put method should contain "Create" or "Update" | [symsAPI.json#L1062](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L1062) |
| Error response should contain a x-ms-error-code header. | [symsAPI.json#L1137-L1141](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L1137-L1141) |
| A delete operation should have a `204` response. | [symsAPI.json#L1206-L1214](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L1206-L1214) |
| All success responses except 202 & 204 should define a response body. | [symsAPI.json#L1207-L1209](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L1207-L1209) |
| Error response should contain a x-ms-error-code header. | [symsAPI.json#L1210-L1214](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L1210-L1214) |
| Operation might be pageable. Consider adding the x-ms-pageable extension. | [symsAPI.json#L1220-L1258](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L1220-L1258) |
| Error response should contain a x-ms-error-code header. | [symsAPI.json#L1254-L1258](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L1254-L1258) |
| Property should have a description. | [symsAPI.json#L1337-L1339](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L1337-L1339) |
| Property should have a description. | [symsAPI.json#L1411-L1413](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L1411-L1413) |
| Property should have a description. | [symsAPI.json#L1515-L1517](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L1515-L1517) |
| Property should have a description. | [symsAPI.json#L1576-L1578](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L1576-L1578) |
| Property should have a description. | [symsAPI.json#L1607-L1609](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L1607-L1609) |
| Property should have a description. | [symsAPI.json#L1641-L1643](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L1641-L1643) |
| Property should have a description. | [symsAPI.json#L1980-L1984](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L1980-L1984) |
| Property should have a description. | [symsAPI.json#L1995-L2000](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L1995-L2000) |
| Error response schema should contain an object property named `error`. | [symsAPI.json#L2133-L2137](https://www.github.com/Azure/azure-rest-api-specs/blob/24956332454abd4990f4cd5c7907e219710d8ff7/specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json#L2133-L2137) |

File: specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json [hunk 1]
```diff
@@ -0,0 +1,2200 @@
...
+        },
+        "target": {
+          "description": "Property name/path in request associated with error.",
+          "type": "string"
+        },
+        "details": {
+          "description": "Array with additional error details.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CloudError"
```
Comment (weidongxu-microsoft) [inline specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json:2164]: This reference likely be wrong. It should ref to `CloudErrorBody`. See https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#Handling-Errors

File: specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json [hunk 2]
```diff
@@ -0,0 +1,2200 @@
...
+      "description": "The Synapse client API Version."
+    },
+    "UserAgentParameter": {
+      "name": "User-Agent",
+      "type": "string",
+      "in": "header",
+      "required": true,
+      "x-ms-parameter-location": "client",
+      "description": "User-Agent request header for servers to identify application"
+    },
```
Comment (weidongxu-microsoft) [inline specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json:2189]: Does synapse have specific requirement on user-agent provided by customer? What if user call REST without user-agent or just CURL user-agent.

File: specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json [hunk 3]
```diff
@@ -0,0 +1,2200 @@
...
+          "type": "array",
+          "items": {
+            "description": "MDEntity (Database/Table/Schema/View/PartitionInfo/PermissionAssignment",
+            "type": "object"
+          }
+        },
+        "continuationToken": {
+          "description": "Continuation token to get next page.",
+          "type": "string"
+        }
```
Comment (weidongxu-microsoft) [inline specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json:2005]: Standard pagination should use opaque "nextLink"

File: specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json [hunk 4]
```diff
@@ -0,0 +1,2200 @@
...
+              "TABLE",
+              "SCHEMA",
+              "VIEW",
+              "FUNCTION",
+              "PARTITIONINFO",
+              "RELATIONSHIP"
+            ],
+            "x-ms-enum": {
+              "name": "SASEntityType",
+              "modelAsString": false
```
Comment (weidongxu-microsoft) [inline specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json:300]: modelAsString=false means you cannot add new enum value, ever.

File: specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json (hunk 2 above)
Comment (piyushdubey) [reply specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json:2189]: No specific requirements, CURL requests without User-Agent will fail. 
The CURL request will need to have a user-agent specified with -A for it to go thorugh successfully.

File: specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json (hunk 1 above)
Comment (piyushdubey) [reply specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json:2164]: This model is exactly the same as what Synapse is already using.
Refer - https://github.com/Azure/azure-rest-api-specs/blob/main/specification/synapse/common/v1/types.json

Comment (weidongxu-microsoft) [reply specification/synapse/data-plane/Microsoft.Synapse/stable/2021-04-01/symsAPI.json:2164]: Then why do not refer to that object in common/v1/types.json?

**Pull request closed without merging**